### ISSUE TITLE
Metadata in CLI?

### ISSUE THREAD

**Issue by hako** [onyxbits/Raccoon#18]
I love using Raccon, but It would be great if it can output metadata (based on the 'Details' section) somehow in the CLI for developers to work with.

**Comment by onyxbits** [comment]
Please give some more detail (what is needed, in what format?), but
don't provide patches. There won't be any more v3.x releases.

Am Freitag, den 07.08.2015, 13:55 -0700 schrieb Wesley Hill:

**Comment by hako** [comment]
Say for instance, If I would like to get the details of the Android app in the CLI. I could do this:

` java -jar raccoon.jar -m com.facebook.katana`
`-m` could mean metadata.

The output of that command is a JSON or XML representation of the details page in the Raccon GUI.

![facebook](https://cloud.githubusercontent.com/assets/2040416/9195444/d32792fe-401a-11e5-9d65-c15c2bc7826b.png)

Example JSON output:

details.json
```json
{
 "name":"Facebook", 
 "package":"com.facebook.katana", 
 "size":"43.5MB", 
 "published":"August 10, 2015", 
 "price":"Free",
 "version":"40.0.0.24.199",
 "raw_version":13607412,
 "installations":"100,000,000+", 
 "rating":4.00
}
```

**Comment by onyxbits** [comment]
That would be relatively easy to do. The problem just is the output
format. You want JSON, someone else might want XML, yet another one
wants SQL. I'm afraid this will go down a road where we get a huge
amount of (slightly) different output formats and a splitting headache
trying to maintain them all.

Am Dienstag, den 11.08.2015, 04:11 -0700 schrieb Wesley Hill:

**Comment by IzzySoft** [comment]
Using templates with place-holders might be an option here – so the user is completely free to define even his/her own format.

**Comment by onyxbits** [comment]
I would need to see a complete proposal there. I'm not going to guess
what might be needed and then implement something nobody ever uses.

Am Donnerstag, den 19.11.2015, 15:37 -0800 schrieb Izzy:

**Comment by IzzySoft** [comment]
Agreed. I was just throwing in an idea – maybe @hako can build upon that. I'm afraid the needs will differ pretty much: while one only needs "core data", another might want a "full dump" (if I wanted to use that, I'd need the full package data including permissions).
